pkg/apis/core/validation/validation.go

```diff
@@ -2178,6 +2181,10 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 		allErrs = append(allErrs, ValidateImmutableAnnotation(newPvc.ObjectMeta.Annotations[v1.BetaStorageClassAnnotation], oldPvc.ObjectMeta.Annotations[v1.BetaStorageClassAnnotation], v1.BetaStorageClassAnnotation, field.NewPath("metadata"))...)
 	}
 
+	if !validateStorageClassUpgradeFromNil(oldPvcClone.Annotations, oldPvcClone.Spec.StorageClassName, newPvcClone.Spec.StorageClassName, opts) {
+		allErrs = append(allErrs, ValidateImmutableField(newPvc.Spec.StorageClassName, oldPvc.Spec.StorageClassName, field.NewPath("metadata").Child("storageClassName"))...)
```

Comment:
This results in a "field is immutable" error, but since the PR establishes conditions where it is mutable, should this be a `field.Invalid` with a bit more of an explanation?

Reply:
Good question.  For node PodCIDR we return `field.Forbidden` for "sometimes allowed but not like this".  That's not clearly right either.

Not sure which is best here, but Immutable seems wrong..

Reply:
I haven't found any simple way to return more specific error yet. Added a TODO note to fix it during beta.